src/poetry/utils/env.py

```diff
@@ -1839,6 +1842,46 @@ def ephemeral_environment(
         yield VirtualEnv(venv_dir, venv_dir)
 
 
+@contextmanager
+def build_environment(
+    poetry: P, env: Env | None = None, io: IO | None = None
+) -> Iterator[Env]:
+    """
+    If a build script is specified for the project, there could be additional build
+    time dependencies, eg: cython, setuptools etc. In these cases, we create an
+    ephemeral build environment with all requirements specified under
+    `build-system.requires` and return this. Otherwise, the given default project
+    environment is returned.
+    """
+    if not env or poetry.package.build_script:
+        with ephemeral_environment(executable=env.python if env else None) as venv:
```

Comment:
Pretty sure this `executable=` will do nothing as you just checked `not env`.

Reply:
There can be a case where env is none and a build script exists.

Reply:
Ah, yeah, I read the `or` as a `and` -- oops.